File: Top_interview_150/14-Longest-Common-Prefix.cpp
class Solution {
public:
    string compare(string str1,string str2){
        int shortest = min(str1.size(), str2.size());
        int i = 0;
        while(i<shortest && str1[i]==str2[i])i++;
        return str1.substr(0,i);
    }
    string longestCommonPrefix(vector<string>& strs) {
        string longest_prefix=strs[0];
        for(int i=1;i<strs.size();i++){
            longest_prefix=compare(longest_prefix,strs[i]);
            if(longest_prefix.empty())return \\;
        }
        return longest_prefix;
    }
};

File: Random/Strings/14-Longest-Common-Prefix.ts

function longestPrefix(str1:string,str2:string):string{
  let len1=str1.length,len2=str2.length;

    let prefix:string = "";

    for(let i=0;i<len1&&i<len2;i++){
        if(str1[i]===str2[i])prefix+=str1[i];
        else break;
    }
return prefix;
}

function longestCommonPrefix(strs: string[]): string {
    const len=strs.length;
  let result=strs[0];
  for(let i=1;i<len;i++){
    result=longestPrefix(strs[i],result);
  }
  return result;
};

